Laravel 10 support

bennothommo opened the pull request:
WIP PR for Laravel 10 support.

bennothommo committed:
Update Composer deps with dev L10 branches, retarget tests

bennothommo committed:
Merge branch 'develop' into wip/laravel-10

bennothommo committed:
Update doctrine/dbal dependency

bennothommo committed:
Match publicPath signature with Laravel

bennothommo committed:
Rename route middleware property

Laravel has renamed it in Laravel 10, although the old name still works. Better to get on top of it now since they'll probably drop the old name at some point.

Refs: https://laravel.com/docs/10.x/upgrade#middleware-aliases

bennothommo committed:
Re-implement $dates model property support

LukeTowers commented on composer.json:
```diff
@@ -36,10 +36,10 @@
-        "laravel/framework": "^9.1",
-        "laravel/tinker": "^2.7",
+        "doctrine/dbal": "^3.0",
+        "enshrined/svg-sanitize": "^0.15",
```
Why was this bumped down?

bennothommo replied:
Must've missed this in the branch merge. Will adjust.

LukeTowers committed:
Merge branch 'develop' into wip/laravel-10

# Conflicts:
#	.github/workflows/tests.yml

LukeTowers committed:
Bump svg-sanitize

LukeTowers committed:
Pass the mocked MailManager to the MailFake for testing

Refs: https://github.com/laravel/framework/commit/2c51b48615f4fa228767dc9b536f6404def4ed41 & https://github.com/laravel/framework/commit/48ed0eb6524b261daaddad1243b24641e02220d4

LukeTowers committed:
Add missing import

LukeTowers committed:
Support multiple paths in the translation file loader

See https://github.com/laravel/framework/pull/45835

LukeTowers committed:
Fix PHP depreciations

LukeTowers committed:
Fix support for Winter style locales (xx-xx)

See https://github.com/wintercms/storm/pull/137/commits/19be709afc9879b47084937d81810c7eb1919fcb & https://github.com/laravel/framework/pull/45835

LukeTowers committed:
Account for vendor override loading support in Translation FileLoader test

Also fix variable names in Translation FileLoader

LukeTowers committed:
Merge branch 'develop' into wip/laravel-10

LukeTowers committed:
Support symfony/console 6.3+

LukeTowers committed:
Laravel added new forget() method to MessageBag objects

Added in https://github.com/laravel/framework/commit/0068c43a7bec4d135cc0b88f4e6847d54b7ddb8c as remove(), renamed to forget() in https://github.com/laravel/framework/commit/47190cc9f51e5f571a47ff4e6873917f3bcf814e.

LukeTowers committed:
Fix support for the Mail::fake() method

See https://github.com/wintercms/storm/pull/137/commits/82d2dfd29e5da36982de67a6334ef74ec4ae1b1d

LukeTowers committed:
Override Laravel's CacheManager to improve static analysis

The tags() method is present on the Repository class but not on the contract. It is also present on all cache drivers supported out of the box because they all extend the base Repository class.

LukeTowers committed:
Static analysis fixes

bennothommo committed:
Merge branch 'develop' into wip/laravel-10

bennothommo committed:
Merge branch 'wip/laravel-10' of github.com:wintercms/storm into wip/laravel-10

LukeTowers commented:
@bennothommo all of the core cache repositories provided by Laravel extend the base Repository class, even the ones that don't support tags, so I'm fine with type hinting it instead.

mjauvin committed:
Merge branch 'wip-laravel-11' into wip/laravel-10
